src/BCS.h

```diff
@@ -0,0 +1,230 @@
+// Copyright © 2017-2022 Trust Wallet.
+// Created by Clément Doumergue
+
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include <algorithm>
+#include <concepts>
+#include <iterator>
+#include <optional>
+#include <string_view>
+#include <tuple>
+#include <variant>
+
+#include "Data.h"
+
+namespace TW::BCS {
+
+/// Implementation of BCS encoding (as specified by the Diem project, see github.com/diem/bcs#detailed-specifications)
+
+struct output_stream {
```

Comment:
Let's use upper case `Serializer` here
```suggestion
struct Serializer {
```